suite/tests/client-interface/drx-scattergather-aarch64.cpp

```diff
@@ -398,12 +467,56 @@ run_tests(const std::vector<scalar_plus_vector_test_case_t> &tests)
     return overall_status;
 }
 
-class input_data_t {
+class test_memory_t {
 public:
     const size_t DATA_SIZE = 9 * 4096; // 9 4KiB pages
-    input_data_t()
+    test_memory_t()
         : data(mmap(nullptr, DATA_SIZE, PROT_READ | PROT_WRITE,
                     MAP_PRIVATE | MAP_ANONYMOUS, -1, 0))
+    {
+        reset();
+    }
+
+    ~test_memory_t()
+    {
+        munmap(data, DATA_SIZE);
+    }
+
+    void
+    reset()
+    {
+        // Remap all the memory read+write so we can write the poison value.
+        mmap(data, DATA_SIZE, PROT_READ | PROT_WRITE,
+             MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+
+        static constexpr uint8_t POISON_VALUE = 0xAB;
+        memset(data, POISON_VALUE, DATA_SIZE);
+
+        // Change the permissions of pages > 1 so that any accesses to them will fault.
+        mmap(region_start_addr(4), 8 * 4096, PROT_NONE,
+             MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+    }
+
+    const void *
+    region_start_addr(size_t offset) const
+    {
+        return &static_cast<char *>(data)[1024 * offset];
```

Comment:
How about a named constant for this 1024 used in several places.

Reply:
Done.